python/graphstorm/gconstruct/remap_result.py

```diff
@@ -0,0 +1,477 @@
+            logging.debug("worker %d Processing %s task", worker_id, idx)
+            func(**task_args)
+    except queue.Empty:
+        pass
+
+def multiprocessing_remap(tasks, num_proc, remap_func):
+    """ Do multi-processing remap
+
+        Parameters
+        ----------
+        task: list
+            List of remap tasks.
+        num_proc: int
+            Number of workers to spin up.
+        remap_func: func
+            Reampping function
+    """
+    if num_proc > 1 and len(tasks) > 1:
+        if num_proc > len(tasks):
+            num_proc = len(tasks)
+        processes = []
+        manager = multiprocessing.Manager()
+        task_queue = manager.Queue()
+        for i, task in enumerate(tasks):
+            task_queue.put((i, task))
+
+        for i in range(num_proc):
+            proc = Process(target=worker_fn, args=(i, task_queue, remap_func))
+            proc.start()
+            processes.append(proc)
+
+        for proc in processes:
+            proc.join()
+    else:
+        for i, task_args in enumerate(tasks):
+            logging.debug("worker 0 Processing %s task", i)
+            remap_func(**task_args)
+
+def _get_file_range(num_files, rank, world_size):
+    """ Get the range of files to process by the current instance.
+
+        Parameters
+        ----------
+        num_files: int
+            Total number of files.
+        rank: int
+            Current worker rank.
+        world_size: int
+            Total number of workers.
+
+        Return
+        ------
+        (int, int): start idx and end idx
+    """
+    if world_size == 0:
+        return 0, num_files
+
+    files_per_inst = num_files // world_size
+    rest = num_files % world_size
+    # evenly spread the fiels
+    # For example, with 10 files and world_size == 4
+    # number files for each rank will be:
+    # 2 2 3 3
+    if rank + rest < world_size:
+        start = files_per_inst * rank
+        end = start + files_per_inst
+    else:
+        start = files_per_inst * rank + rank + rest - world_size
+        end = start + files_per_inst + 1
+
+    return start, end
+
+def remap_edge_pred(pred_etypes, pred_dir,
+                    output_dir, out_chunk_size,
+                    num_proc, rank, world_size, with_shared_fs,
+                    preserve_input=False):
+    """ Remap edge prediction result.
+
+        The function wil iterate all the edge types that
+        have prediction results and spin num_proc workers
+        to do the rampping jos.
+
+        The directory storing prediction results looks like:
+        # Predicionts:
+        #    predict-00000.pt
+        #    predict-00001.pt
+        #    ...
+        #    src_nids-00000.pt
+        #    src_nids-00001.pt
+        #    ...
+        #    dst_nids-00000.pt
+        #    dst_nids-00001.pt
+        #    ...
+
+        The output emb files will be
+        #    predict-00000_00000.parquet
+        #    predict-00000_00001.parquet
```

Comment:
why the remapped data are stored in a different format?

Reply:
We can hardly store strings in torch tensor.
We supports parquet now. But we can extend it into csv/json or other formats if needed.

Reply:
why can we not store them in torch tensors?

Reply:
Does torch support string tensors?
I think parquet is a general format that can be easily accepted by many downstream applications.